packages/rnpm-plugin-windows/src/common.js

```diff
@@ -41,61 +41,88 @@ function getLatestVersion() {
-      }
+function getLatestMatchingVersion(versionRange, tag) {
+  return new Promise((resolve, reject) => {
+    // Ignore the version range of React Native if asking for master, since our
+    // RNW version may not align.
+    if (tag === 'master') {
+      npm.packages.release('react-native-windows', 'master', (err, rel) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(rel[0].version);
+        }
+      });
+    } else {
+      npm.packages.releases('react-native-windows', (err, rels) => {
+        if (err) {
+          reject(err);
+        } else {
+          const matchingVersions = Object.keys(rels)
+            .filter(v => isVersionMatch(v, versionRange, tag))
+            .sort(semver.rcompare);
 
-      const matchedVersion = release.version;
-      const matchedPrerelease = semver.prerelease(matchedVersion);
-      const isPrerelease = tag && !!matchedPrerelease;
-      if (!isVersionMatch(matchedVersion, version, tag) && (ignoreStable || isPrerelease)) {
-        const versions = Object.keys(release.versions);
-        const candidates = versions.filter(v => isVersionMatch(v, version, tag)).sort(semver.rcompare);
-        if (candidates.length === 0) {
-          const tagMatches = versions.filter(v => isTagMatch(v, tag)).sort(semver.rcompare);
-          if (tagMatches.length === 0) {
-            reject(new Error(`Could not find react-native-windows@${version}-${tag}.*.`));
+          if (matchingVersions.length > 0) {
+            resolve(matchingVersions[0]);
           } else {
-            const latestVersion = tagMatches[0];
-            reject(new Error(`Could not find react-native-windows@${version}-${tag}.*. ` +
-              `Latest version of react-native-windows for tag '${tag}' is ${latestVersion}, try switching to ` +
-              `react-native@${semver.major(latestVersion)}.${semver.minor(latestVersion)}.*.`));
+            reject();
           }
         }
-        resolve(candidates[0]);
-      } else {
-        resolve(matchedVersion);
-      }
-    });
+      });
+    }
   });
 }
 
-const getInstallPackage = function (version, tag, useStable) {
+async function getMatchingVersion(versionRange, tag) {
+  const versionStr = tag === 'master' ? 'master' : versionRange;
+  console.log(`Checking for react-native-windows version matching ${versionStr}...`);
+
+  try {
+    return await getLatestMatchingVersion(versionRange, tag);
+  } catch (ex) {
+    const latestVersion = await getLatestVersion();
+    throw new Error(`Could not find react-native-windows@${versionStr}. ` +
+      `Latest version of react-native-windows is ${latestVersion}, try switching to ` +
+      `react-native@${semver.major(latestVersion)}.${semver.minor(latestVersion)}.*.`);
+  }
+}
+
+const isSupportedVersion = function(validVersion, validRange) {
+  // Allow 0.0.0-x master builds
+  if (validVersion) {
+    return semver.lt(validVersion, '0.0.0') || semver.gte(validVersion, '0.27.0');
```

Comment:
>semver.lt(validVersion, '0.0.0') [](start = 11, length = 32)

0.0.0-master.x is < 0.0.0?

Reply:
The former is a pre-release, so it should be. I think I tested this case, but can double check.

Reply:
yes 0.0.0-master.x should be < 0.0.0

Reply:
Thanks!  I fin this bizarre, but this isn't the first time I've seen something bizarre with npm version rules.